lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -40,6 +43,24 @@ def lookup(path, vault_url = nil)
 
   private
 
+  def create_ssl_context(verify_ssl)
+    ssl_provider = Puppet::SSL::SSLProvider.new
+    default_ssl_context = ssl_provider.load_context()
```

Comment:
In another project I used `Puppet.lookup(:ssl_context)`. Looking at https://github.com/puppetlabs/puppet/blob/efa75f0fb0b5d4877750c3eadddd70bcc4ee081d/lib/puppet.rb#L210-L223 it looks like that looks like it also handles the password and might be a better base.